### PR TITLE
Add filter flag to top pages

### PR DESCRIPTION
**Pull request by koos42**
Adds support for filtering the `topPages` call to linkscape api.

### CODE REVIEW

File: spec/integration/linkscape/client_spec.rb [hunk 1]
```diff
@@ -4,9 +4,12 @@
 require 'yaml'
 require 'linkscape'
 
+require 'pp'
+require 'debugger'
```
Comment (bkirz) [inline spec/integration/linkscape/client_spec.rb]: Debug code.

File: spec/integration/linkscape/client_spec.rb [hunk 2]
```diff
@@ -20,10 +23,40 @@ module Linkscape
...
+            :sort => sort,
+            :limit => 4,
+            :offset => 0,
+            :filter => 'status4xx'
+          }
+        response = client.topPages(uri, uri_type, params)
+        statuses = response.data.map { |d| d[:source][:status] }
+        expect(statuses.size).to be > 0
+        statuses.each { |s| expect(s).to be_between(400, 499) }
+      end
```
Comment (bkirz) [inline spec/integration/linkscape/client_spec.rb:62]: Just had an idea: what do you think about making requests with two different filters? That should increase our confidence that the filter is actually working (as opposed to it not working but incidentally returning correct results).

**Comment by bkirz** [comment]
I don't think committing the re-recorded cassettes for the index fetcher is necessary.

**Comment by bkirz** [comment]
LGTM once the comments are addressed.

File: spec/integration/linkscape/client_spec.rb (hunk 2 above)
Comment (koos42) [inline spec/integration/linkscape/client_spec.rb:62]: sounds good

**Pull request merged**